examples/lb3-application/src/__tests__/acceptance/lb3app.acceptance.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-lb3-application
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {OpenApiSpec} from '@loopback/rest';
+import {Client, expect} from '@loopback/testlab';
+import * as _ from 'lodash';
+import {CoffeeShopApplication} from '../../application';
+import {givenCoffeeShop, setupApplication} from './test-helper';
+
+const {generateSwaggerSpec} = require('loopback-swagger');
+const swagger2openapi = require('swagger2openapi');
+
+const lb3App = require('../../../lb3app/server/server');
+
+describe('CoffeeShopApplication', () => {
+  let app: CoffeeShopApplication;
+  let client: Client;
+
+  before('setupApplication', async () => {
+    ({app, client} = await setupApplication());
+  });
+
+  after('closes application', async () => {
+    if (app) await app.stop();
+  });
+
+  context('basic REST calls for LoopBack 3 application', () => {
+    it('creates and finds a CoffeeShop', async () => {
+      const coffeeShop = givenCoffeeShop();
+      const response = await client
+        .post('/api/CoffeeShops')
+        .send(coffeeShop)
+        .expect(200);
+
+      expect(response.body).to.containDeep(
+        _.pick(coffeeShop, ['name', 'city']),
+      );
+
+      const result = await client.get(`/api/CoffeeShops/${response.body.id}`);
+      expect(result.body).to.containDeep(_.pick(coffeeShop, ['name', 'city']));
+    });
+
+    it("gets the CoffeeShop's status", async () => {
+      const currentDate = new Date();
+      const currentHour = currentDate.getHours();
+      const response = await client.get('/api/CoffeeShops/status').expect(200);
+
+      if (currentHour >= 6 && currentHour < 20) {
```

Comment:
This makes the test brittle and prone to timing errors. Consider this timing:
 - in the test, we observe the current date as `19h59m59.999s`
 - by the time the request reaches the status endpoint, current date becomes `20h00m00.001s`

A better solution: use a fake timer and force `new Date` to always return the same value you want. See https://sinonjs.org/releases/v7.3.2/fake-timers/

Alternatively, I think it's ok to simplify this test to simply check that the response is one of the two expected strings.

```ts
const response = await client.get('/api/CoffeeShops/status').expect(200);
expect(response.body.status).to.be.equalOneOf(
   'We are open for business.',
  'Sorry, we are closed. Open daily from 6am to 8pm.',
);
```

See http://shouldjs.github.io/#assertion-equaloneof